packages/sdk/src/protocols/light_push.ts

```diff
@@ -43,15 +50,19 @@ class LightPushSDK extends BaseProtocolSDK implements ILightPushSDK {
       };
     }
 
-    const peers = await this.protocol.getPeers();
-    if (!peers.length) {
+    const peersFound = await this.hasPeers();
```

Comment:
it seems to me in this case we should `get` peers as fast as possible and only in background support validity and quality of the pool

because if we make it part of `send` or `subscribe` (or for Store) - then it increases delay for the application built on top

instead I think we should be lazy in this approach and prevent / make consumer wait only if service doesn't work (error being thrown) 

edge case is if the only peer is in the pool and this peer is bad - I believe it is a rare case for which we should optimize only after seeing it happens often for end users.
So we should measure it now: add a log line for it for later use in telemetry etc, also a point for dogfood app we are preparing.

Reply:
> and thus we need to ensure that maintainPeers() was called at least once
it seems it is initiated from the `constructor` so there is no need to make sure of it anymore, considering `ctor` was initiated successfully 

let's not have blocking routines for ad-hoc operations, if we cannot send a message now - we should throw (maybe later we can implement improvement of re-sending messages but this is out of scope)

Reply:
I like the idea of `autoRetry` and we can enrich it within https://github.com/waku-org/js-waku/issues/2032 (feel free to take if you want and when free and let's discuss exact design, we can change it from what I described there)

and I think it is a good way to use `peer:identify` or some other event (maybe `connected` is good for that purpose too)

but I would still not make consumer wait for some other operation during `.send` unless it is explicitly configured